plugins/out_azure/azure_conf.c

```diff
@@ -42,6 +42,9 @@ struct flb_azure *flb_azure_conf_create(struct flb_output_instance *ins,
         return NULL;
     }
     ctx->ins = ins;
+    
+    /* Set default values */
+    flb_output_config_map_set(ins, ctx);
 
     /* config: 'customer_id' */
     cid = flb_output_get_property("customer_id", ins);
```

Comment:
In your proposed version, the configuration map is not fully levaraged,
because we do manual parameter assignments anyway here.

Since `flb_output_config_map_set()` is supposed to take care of the most
of these tasks,  you need to examine the existing parameter handlings one
by one and determine if what we can remove each.

Reply:
I have spent the last one day examining and testing the existing parameters one by one and I came to the conclusion that we cannot remove any. I have come to this based on differrent logic for different parameters. Could we maybe this discuss this ?

Reply:
> I have spent the last one day examining and testing the existing parameters one by one and I came to the conclusion that we cannot remove any.

Can you elaborate this reasoning?

It seems to me that we can remove the assignment of `customer_id` here,
since the exact value is already assigned to `ctx->customer_id` at this point.

Reply:
For customer-id valgrind shows error on removing assignment

Reply:
My bad, I was not understanding earlier. I have made the relevant changes, please review again. Thank you :)